### DISCUSSION TITLE
gettext 0.21 libintl.9.dylib missing MacOS?

### DISCUSSION THREAD

**Discussion by andrewcrook** [Homebrew/discussions#3391] [Everyday usage]
I have been trying to compile an application and its failing because libintl.9.dylib cannot be found 
my system has the following installed via homebrew 

```
gettext: stable 0.21 (bottled)
GNU internationalization (i18n) and localization (l10n) library
https://www.gnu.org/software/gettext/
```
on 

> MacOS 12.4 (21F79)
> M1 MBP

This package contains the for mentioned library, however, version 8 i.e libintl.8.dylib 
The application that I was compiling said its dependency was gettext 0.18.x in the Readme
This is confusing as homebrew's gettext is newer yet doesn’t contain libintl.9.dylib
One of the errors 

```
raco setup: error: during building docs for <pkgs>/redex-doc/redex/redex.scrbl
raco setup:   ffi-lib: could not load foreign library
raco setup:     path: libintl.9.dylib
raco setup:     system error: dlopen(libintl.9.dylib, 0x000A): tried: 'libintl.9.dylib' (no such file), '/usr/local/lib/libintl.9.dylib' (no such file), '/usr/lib/libintl.9.dylib' (no such file), '/private/var/folders/tg/wd1yw6m12mz7vyb1nw2wz9n80000gn/T/racket_XXXXXXX.g4yYFSEv/racket-8.5/src/cs/c/libintl.9.dylib' (no such file)
```

Does anyone know how I can get this working?
where does libintl.9.dylib come from?

```
HOMEBREW_VERSION: 3.5.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: fba051f98f609634ae4392d0053ddfc627f8cdb3
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e156c279eba4cba371af2acdf2ad5c376b4a402c
Core tap last commit: 6 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_AUTO_UPDATE_SECS: 86400
HOMEBREW_CASK_OPTS: ["--require-sha"]
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ANALYTICS: set
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_NO_INSECURE_REDIRECT: set
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.37.0 => /opt/homebrew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.4-arm64
CLT: 13.4.0.0.1.1651278267
Xcode: 13.4.1
Rosetta 2: false
```

**Comment by carlocab** [comment]
> Does anyone know how I can get this working?
> where does libintl.9.dylib come from?

These seem like questions best directed to the developers of the application you're trying to build, since it's their software that is looking for `libintl.9.dylib`.

**Comment by tgbugs** [comment]
I'm seeing this as well. It looks like libintl.9 was changed in https://github.com/racket/racket/commit/5bb2c1f16ed38491f1042bc702366fecdc7c7087 which was back in 2018, so something is strange about this. @mflatt any idea what might have changed that caused this to be fatal now?

**Reply by tgbugs** [reply]
https://racket.discourse.group/t/libintl-9-dylib-cannot-be-verified-error-with-macos-executable/690

related

**Reply by Bo98** [reply]
Looks like that's using the known bugged gettext 0.19.8 release. 0.19.8.1 was released 2 days later to fix incorrect ABI versions.

**Reply by mflatt** [reply]
A Racket installation on Mac normally includes a number of shared libraries that are installed via packages. A `libintl.9.dylib` should be installed (in Racket's `lib` directory) via the `draw-x86_64-macosx-3` package. The `redex` package depends on that package through transitive dependencies. So, the question is probably one of something going wrong in Racket package installation.

Possibly related, although it's not clear: racket/racket#4334.

**Reply by andrewcrook** [reply]
It's not fatal and a lot has changed since.

The original post was because libintl.9.dylib doesn’t actually exist in gettext. libintl.9.dylib was in fact libintl.8.dylib
It was a versioning mistake which has been resolved in gettext. 

The only way to correct this is to get the Racket team to use an updated version of gettext so its back to libintl.8.dylib.
Racket team aren't really in favour of updating the dylibs especially when it still works.
 
As stated Racket does actually install them via certain packages the issue I was having was one compiling Racket from source not the racket brew packages. That just takes some understanding of the packages and source. Some of it wasn't clear at the time.

At first I thought it was an issue with brew gettext package, however, it isn't.

**Reply by andrewcrook** [reply]
> https://racket.discourse.group/t/libintl-9-dylib-cannot-be-verified-error-with-macos-executable/690
> 
> related

That is a gatekeeper issue. 
The developer needs to codesign and notarise his macOS app on a Mac via an Apple developer account.

**Comment by tgbugs** [comment]
To follow up on my end, it seems that the issue was trying to build racketcgc from the full racket source tarball instead of the minimal racket tarball. The minimal racket version does not appear to evoke the issue.
